File: api/src/test/java/net/blueshell/api/factory/dto/PictureDTOFactory.java
package net.blueshell.api.factory.dto;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.PictureDTO;
import org.springframework.stereotype.Component;

/**
 * Factory for PictureDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class PictureDTOFactory extends BaseDtoFactory<PictureDTO> {

    @Override
    public Class<PictureDTO> targetType() {
        return PictureDTO.class;
    }

    @Override
    public PictureDTO createBasic() {
        PictureDTO dto = new PictureDTO();
        dto.setName(unique("picture"));
        dto.setUrl("https://cdn.example.com/" + unique("pic") + ".jpg");
        dto.setUploaderId(nextId());
        dto.setEventId(nextId());
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/PictureDTOFactory.kt
package net.blueshell.api.factory.dto

import net.blueshell.api.dto.PictureDTO
import org.springframework.stereotype.Component

/**
 * Factory for PictureDTO test instances.
 */
@Component
class PictureDTOFactory : BaseDtoFactory<PictureDTO>() {

    override fun targetType(): Class<PictureDTO> = PictureDTO::class.java

    override fun createBasic(): PictureDTO {
        val dto = PictureDTO()
        dto.name = unique("picture")
        dto.url = "https://cdn.example.com/${unique("pic")}.jpg"
        dto.uploaderId = nextId()
        dto.eventId = nextId()
        return dto
    }
}
